conda-store-server/conda_store_server/plugins/plugin_manager.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) conda-store development team. All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+import pluggy
+
+from conda_store_server.exception import CondaStorePluginNotFoundError
+from conda_store_server.plugins import BUILTIN_PLUGINS, types
+
+
+class PluginManager(pluggy.PluginManager):
+    def get_lock_plugins(self) -> dict[str, types.TypeLockPlugin]:
+        """Returns a dict of lock plugin name to class"""
+        plugins = [item for items in self.hook.lock_plugins() for item in items]
+        return {p.name.lower(): p for p in plugins}
+
+    def lock_plugin(self, name: str) -> types.TypeLockPlugin:
+        """Returns a lock plugin by name"""
+        lockers = self.get_lock_plugins()
+
+        if name not in lockers:
+            raise CondaStorePluginNotFoundError(
+                plugin=name, available_plugins=lockers.keys()
+            )
+
+        return lockers.get(name)
```

Comment:
```suggestion
        return lockers[name]
```

This is guaranteed to exist now.